alvr/client_core/src/platform/android/decoder.rs

```diff
@@ -242,19 +259,51 @@ pub fn video_decoder_split(
                 .set_buffer_removed_listener(Box::new(|_, _| ()))
                 .unwrap();
 
-            let decoder = Arc::new(FakeThreadSafe(
-                MediaCodec::from_decoder_type(&mime).unwrap(),
-            ));
+            let mime = match config.codec {
+                CodecType::H264 => "video/avc",
+                CodecType::Hevc => "video/hevc",
+            };
+
+            let sw_codec_name = match config.codec {
+                CodecType::H264 => "OMX.google.h264.decoder",
+                CodecType::Hevc => "OMX.google.hevc.decoder",
+            };
+
+            let format = MediaFormat::new();
+            format.set_str("mime", mime);
+            format.set_i32("width", config.width);
+            format.set_i32("height", config.height);
+            format.set_buffer("csd-0", &csd_0);
+
+            for (key, value) in &config.options {
+                match value {
+                    MediacodecDataType::Float(value) => format.set_f32(key, *value),
+                    MediacodecDataType::Int32(value) => format.set_i32(key, *value),
+                    MediacodecDataType::Int64(value) => format.set_i64(key, *value),
+                    MediacodecDataType::String(value) => format.set_str(key, value),
+                }
+            }
 
             info!("Using AMediaCoded format:{} ", format);
-            decoder
-                .configure(
-                    &format,
-                    Some(&image_reader.window().unwrap()),
-                    MediaCodecDirection::Decoder,
-                )
-                .unwrap();
-            decoder.start().unwrap();
+
+            let preparing_decoder = if config.force_software_decoder {
+                decoder_try_software(sw_codec_name, &format, &image_reader).unwrap()
```

Comment:
The code is within a thread. I'm not sure where it'd go. I am cleaning up to reduce the amount of unwrapping, but there still will be "an unwrap". In addition, trying to de-unwrapify the code results in loss of which specific call failed, which could be bad for debugging.

Reply:
You could use the function `alvr_common::show_err()` to print the error. The error per se might not be interesting, but otherwise the unwrap() will crash the thread silently, which is worse.